gaeguli/pipeline.c

```diff
@@ -512,6 +486,43 @@ _build_vsrc_pipeline (GaeguliPipeline * self, GaeguliVideoResolution resolution,
   return FALSE;
 }
 
+static void
+_set_stream_caps (GaeguliPipeline * self, GaeguliVideoResolution resolution)
+{
+  gint width, height;
+  g_autoptr (GstElement) capsfilter = NULL;
+  g_autoptr (GstCaps) caps = NULL;
+
+  switch (resolution) {
+    case GAEGULI_VIDEO_RESOLUTION_640X480:
+      width = 640;
+      height = 480;
+      break;
+    case GAEGULI_VIDEO_RESOLUTION_1280X720:
+      width = 1280;
+      height = 720;
+      break;
+    case GAEGULI_VIDEO_RESOLUTION_1920X1080:
+      width = 1920;
+      height = 1080;
+      break;
+    case GAEGULI_VIDEO_RESOLUTION_3840X2160:
+      width = 3840;
+      height = 2160;
+      break;
+    default:
+      width = -1;
+      height = -1;
+      break;
+  }
+
+  caps = gst_caps_new_simple ("video/x-raw", "width", G_TYPE_INT, width,
```

Comment:
Not for this commit, but we have to support `image/jpeg` for higher resolution for 4K usb camera.